core/src/main/java/me/lokka30/treasury/plugin/core/utils/PluginVersion.java

```diff
@@ -12,25 +12,33 @@
 public final class PluginVersion {
 
     private final short majorRevision, minorRevision, patchRevision;
-    private final boolean developmentVersion;
-    private final String stringVersion;
+    private final boolean developmentVersion, release;
+    private final String commit, stringVersion;
 
     public PluginVersion(@NotNull String input, @NotNull Logger logger) {
         Objects.requireNonNull(input, "input");
         Objects.requireNonNull(logger, "logger");
         this.stringVersion = input;
-        if (input.endsWith("-SNAPSHOT")) {
-            developmentVersion = true;
-            input = input.replace("-SNAPSHOT", "");
-        } else {
-            developmentVersion = false;
+
+        if (input.indexOf('.') == -1 || input.indexOf('-') == -1) {
+            throw new IllegalArgumentException(
+                    "Illegal Treasury version found! If this is a live environment version or you're a developer doing changes, please fix this ASAP!");
         }
-        if (input.indexOf('.') == -1) {
-            throw new IllegalArgumentException("Illegal Treasury version found! " + "If this is a live environment version or you're a developer doing changes, please fix this ASAP!");
+        // parse commit before version
+        String[] bigParts = input.split("-");
+        if (bigParts.length != 3) {
+            throw new IllegalArgumentException(
+                    "Illegal Treasury version found! If this is a live environment version or you're a developer doing changes, please fix this ASAP!");
```

Comment:
Perhaps since users will see this, I guess it's better to say `Your current Treasury version is not formatted correctly! Please inform a Treasury maintainer as soon as possible.`

Reply:
(This request also applies to the other usages of `Illegal Treasury version found` throughout this PR)